partiql-planner/src/main/kotlin/org/partiql/planner/catalog/Catalogs.kt

```diff
@@ -0,0 +1,29 @@
+package org.partiql.planner.catalog
+
+/**
+ * Catalogs is used to provide the default catalog and possibly others by name.
+ */
+public interface Catalogs {
```

Comment:
Does this need to be public?

Reply:
Yes, the idea here is to abstract the `Map<String, ConnectorMetadata>` from the session so that a customer may implement their own catalog provider. I understand that you can have a default impl (which is really just a map) that you construct with the planner builder, but consider something like `Catalogs` backed by a DynamoDB implementation.